src/transaction/log_applier_sql_log.c

```diff
@@ -642,6 +686,10 @@ create_dir (const char *new_dir)
 	{
 	  if (mkdir (path, 0777) < 0)
 	    {
+	      er_stack_push ();
+	      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_ES_INVALID_PATH, 1, new_dir);
```

Comment:
You pointed out the right thing. But I really don't want to add new error code. OTL

I will change it.

Reply:
How about use ER_HA_GENERIC_ERROR? SQL logger already use this like "HA generic: Failed to initialize SQL logger.".

Reply:
If only the debug logs are well written, I think it would be okay to use ER_HA_GENERIC. If ER_HA_GENERIC is used, the return value below should also be changed. (return ER_HA_GENERIC_ERROR)

Also, in the function that calls create_dir, it does not check whether the directory has been created (in sl_init ()). I think it would be better to add the code to check if the directory has been created. 
```
if (create_dir (tmp_log_path) != NO_ERROR) ...
```

However, after checking other codes in log_applier_sql_log.c, 
it seems that the `sl_* ` functions do not set error code when error occur. 
I think it is appropriate to set the error where it occurs, but it seems that they all return ER_FAILED and then er_set is done in the caller. 
In this situation, I am wondering whether it is appropriate to only use `er_log_debug` to write debugging logs, or to set the error using `er_set`. What do you think?

Reply:
When ha_sql_log_path is set, the failure of create_dir() is checked in the sl_read_catalog(). So, the error logs are written like:
```
Path for external storage '/root/sql_log' is invalid.
Cannot open SQL catalog file: Permission denied
HA generic: Failed to initialize SQL logger.
```
But, it would be better add an error checking code at the create_dir().

And your opinion looks like every error case should have explict error code. right? I don't aggree with it. Some error situations like logical operation(create dir) is considered as one error code even if there are many reason for the failure like invalid path, permission or non-directory. It seems to depend on how detailed the problem situation is to be shown depending on the importance of the error.